desktop/core/src/desktop/api2.py

```diff
@@ -813,6 +814,44 @@ def export_documents(request):
     return make_response(f.getvalue(), 'json', filename)
 
 
+def topologicalSort(docs):
+
+  size = len(docs)
+  graph = defaultdict(list)
+  for doc in docs:
+    dep_size = len(doc['fields']['dependencies'])
+    for i in range(dep_size):
+      graph[(doc['fields']['dependencies'])[i][0]].append(doc['fields']['uuid'])
+
+  visited = {}
+  _doc = {}
+  for doc in docs:
+    _doc[doc['fields']['uuid']] = doc
+    visited[doc['fields']['uuid']] = False
```

Comment:
Possible to merge the `for` loops?

Reply:
Yes, possible. But for sake of clarity I putted them separate, I am adding comments what task we are performing with this loop.